### ISSUE TITLE
Register with phone number is not working

### ISSUE THREAD

**Issue by peyman1372** [cooperlyt/keycloak-phone-provider#24]
Hi,
the register with a phone number does not work and it just shows the same form click on submit; however, if I set the email as username it does work! tested on keycloak 19.0.0, 19.0.1,19.0.3

**Comment by peyman1372** [comment]
Update, removing profile validation from the conditions, fixed the problem!

**Comment by imcyee** [comment]
I got the same issue too, this issue arise because keycloak expecting `username` to be not null, it works for emailAsUsername because in `DefaultAttributes.java` under `normalizeAttributes` function attributes is auto copied.

 ```
if (email != null && realm.isRegistrationEmailAsUsername()) {
            final List<String> lowerCaseEmailList = email.stream()
                    .filter(Objects::nonNull)
                    .map(String::toLowerCase)
                    .collect(Collectors.toList());

            newAttributes.put(UserModel.USERNAME, lowerCaseEmailList);
        }
```

which doesnt happen to `isRegistrationPhoneAsUsername` under `REGISTRATION_PROFILE` context. The validator requires username to be defined (under AbstractUserProfileProvider).

I believe removing `Profile Validation` might cause missing some data updates as shown in `DefaultUserProfile` under `updateInternal`.

**Comment by cooperlyt** [comment]
you need config "Registration Phone User Creation > setting > Phone number as username > checked " in flows

**Comment by cooperlyt** [comment]
> invalid_registration

I can't reproduce this . what`s your Keycloak version?

**Comment by imcyee** [comment]
@cooperlyt Do you have preview feature: `user profile` enabled?, and also have to enable in the `realm setting`.

**Comment by cooperlyt** [comment]
Tip: if Realm parameter 'Email as username' is true, then config 'Phone number as username' and 'hide email' is invalid! if parameter 'duplicate-phone' is true then 'Phone number as username' is invalid!

confirm this!

**Comment by MBayoumi** [comment]
i have same issue even when using docker compose attached in example

**Comment by cooperlyt** [comment]
fixed it in 2.2.2
